pkg/agent/run.go

```diff
@@ -200,7 +206,11 @@ func Run(cmd *cobra.Command, args []string) {
 			break
 		}
 
-		time.Sleep(config.Period)
+		select {
+		case <-gctx.Done():
+			return
+		case <-time.After(config.Period):
+		}
```

Comment:
If any of the errgroup go routines exit (with nil or error) the main context will be cancelled, which will cause this blocking loop to exit immediately instead of waiting for the time period.